interfaces/builtin/location.go

```diff
@@ -0,0 +1,241 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/ubuntu-core/snappy/interfaces"
+)
+
+var locationPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the location service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+  # DBus accesses
+  #include <abstractions/dbus-strict>
+  dbus (send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member={Request,Release}Name
+     peer=(name=org.freedesktop.DBus),
+
+  dbus (receive, send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member=GetConnectionUnixProcessID
+     peer=(label=unconfined),
+
+  dbus (receive, send)
+     bus=system
+     path=/org/freedesktop/DBus
+     interface=org.freedesktop.DBus
+     member=GetConnectionUnixUser
+     peer=(label=unconfined),
+
+  dbus (send)
+    bus=system
+    path=/org/freedesktop/*
+    interface=org.freedesktop.DBus.Properties
+    peer=(label=unconfined),
+
+  # Allow binding the service to the requested connection names
+  dbus (bind)
+      bus=system
+      name="com.ubuntu.location.Service",
+
+  dbus (bind)
+      bus=system
+      name="com.ubuntu.location.Service.Session",
+
+  # Allow traffic to/from our path and interface with any method
+  dbus (receive, send)
+      bus=system
+      path=/com/ubuntu/location/Service{,/**}
+      interface=com.ubuntu.location.Service*,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}PositionUpdates,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}HeadingUpdates,
+
+  dbus (receive, send)
+      bus=system
+      path=/sessions/*
+      interface=com.ubuntu.location.Service.Session
+      member={Start,Stop}VelocityUpdates,
+
+  # Allow traffic to/from org.freedesktop.DBus for location service
+  dbus (receive, send)
+      bus=system
+      path=/
+      interface=org.freedesktop.DBus**,
+  dbus (receive, send)
+      bus=system
+      path=/com/ubuntu/location/Service{,/**}
+      interface=org.freedesktop.DBus**,
+`)
+
+var locationConnectedPlugAppArmor = []byte(`
+# Description: Allow using location service. Reserved because this gives
+#  privileged access to the service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to location service
+dbus (receive, send)
+    bus=system
+    peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
Yeah, this seems rather dangerous. It assumes we know what will be exposed via this API in the future, which is always a bad bet in terms of security.